Training/etc/di.xml

```diff
@@ -21,4 +21,5 @@
             <argument name="backendSession" xsi:type="object">Magento\Backend\Model\Session\Proxy</argument>
         </arguments>
     </type>
+    <preference for="Codifi\Training\Api\Data\NoteSearchResultInterface" type="Magento\Framework\Api\SearchResults"/>
```

Comment:
Are you sure that you removed right preferences?
`Codifi\Training\Api\Data\NoteSearchResultInterface` doesn't used